In [2]:
import pandas as pd
import numpy as np

In [3]:
df=pd.read_csv("Sales.csv")

In [4]:
df.head(10)

SALESPERSON  Old Scheme (in thousands)  New Scheme (in thousands)
0            1                         57                         62
1            2                        103                        122
2            3                         59                         54
3            4                         75                         82
4            5                         84                         84
5            6                         73                         86
6            7                         35                         32
7            8                        110                        104
8            9                         44                         38
9           10                         82                        107

In [5]:
df.iloc[:,1]=df.iloc[:,1]*1000

In [6]:
df.iloc[:,2]=df.iloc[:,2]*1000

In [7]:
df

SALESPERSON  Old Scheme (in thousands)  New Scheme (in thousands)
0             1                      57000                      62000
1             2                     103000                     122000
2             3                      59000                      54000
3             4                      75000                      82000
4             5                      84000                      84000
5             6                      73000                      86000
6             7                      35000                      32000
7             8                     110000                     104000
8             9                      44000                      38000
9            10                      82000                     107000
10           11                      67000                      84000
11           12                      64000                      85000
12           13                      78000                      99000
13           14                      53000                      39000
14           15                      41000                      34000
15           16                      39000                      58000
16           17                      80000                      73000
17           18                      87000                      53000
18           19                      73000                      66000
19           20                      65000                      78000
20           21                      28000                      41000
21           22                      62000                      71000
22           23                      49000                      38000
23           24                      84000                      95000
24           25                      63000                      81000
25           26                      77000                      58000
26           27                      67000                      75000
27           28                     101000                      94000
28           29                      91000                     100000
29           30                      50000                      68000

#### 1. Mean of New and Old Schemes 

In [8]:
print('Mean of old scheme', df.iloc[:,1].mean()) 

Mean of old scheme 68033.33333333333


In [9]:
print('Mean of new scheme', df.iloc[:,2].mean())

Mean of new scheme 72033.33333333333


In [10]:
Mean_Difference = (df.iloc[:,2].mean() - df.iloc[:,1].mean())
print('Difference in mean', Mean_Difference)

Difference in mean 4000.0


##### 2. It is asked that whether the new scheme has significantly raised the output, it is an example of the one-tailed t-test.


Let 
Mean1 = Average sums assured by salesperson BEFORE changeover. Mean2 = Average sums assured by salesperson AFTER changeover.

NULL Hypothesis: Mean1 = Mean2  ; Mean2 – Mean1  = 0

Alternative Hypothesis: Mean1 < Mean2   ; Mean2 – Mean1  > 0 ; true difference of means is greater than zero.

Since population standard deviation is unknown, paired sample t-test will be used.

#### Using user defined function to calculate the p-value and t_stat values 

In [11]:
from math import sqrt
from numpy.random import seed
from numpy.random import randn
from numpy import mean
from scipy.stats import t

def dependent_ttest(data1, data2, alpha):
    # calculate means
    mean1, mean2 = mean(data1), mean(data2)    
    # number of paired samples
    n = len(data1)
    # sum squared difference between observations
    d1 = sum([(data1[i]-data2[i])**2 for i in range(n)])
    # sum difference between observations
    d2 = sum([data1[i]-data2[i] for i in range(n)])
    # standard deviation of the difference between means
    sd = sqrt((d1 - (d2**2 / n)) / (n - 1))
    # standard error of the difference between the means
    sed = sd / sqrt(n)
    # calculate the t statistic
    t_stat = (mean1 - mean2) / sed
    # degrees of freedom
    degreesoffreedom = n - 1
    # calculate the critical value
    cv = t.ppf(1.0 - alpha, degreesoffreedom)
    # calculate the p-value
    p = (1.0 - t.cdf(abs(t_stat), degreesoffreedom)) * 2.0
    # return everything
    return t_stat, degreesoffreedom, cv, p

t_stat, degreesoffreedom, cv, p = dependent_ttest(df.iloc[:,2], df.iloc[:,1], 0.05)

In [12]:
print('t=%.3f, degreesoffreedom=%d, cv=%.3f, p=%.3f' % (t_stat, degreesoffreedom, cv, p))

t=1.556, degreesoffreedom=29, cv=1.699, p=0.131


#### Cross checking the values obtained above with the scipy.stats ttest_rel function

In [13]:
from scipy.stats import ttest_rel
t_stat, p_stat = ttest_rel(df.iloc[:,2], df.iloc[:,1])
print('t=%.3f, p_stat=%.3f'% (t_stat, p_stat))

t=1.556, p_stat=0.131


####  3. Conclusion: Since p-value (=0.131) is higher than 0.05, we accept (fail to reject) NULL hypothesis. The new scheme has NOT significantly raised outputs.



## 3 ....

### 2. Probability of Type I error = significant level = 0.05 or 5%

##### 3. What is the p-value of the hypothesis test if we test for a difference of $5000?

Solution : 
Let  Mean2 = Average sums assured by salesperson AFTER changeover.

Mean1 = Average sums assured by salesperson BEFORE changeover.

Mean_diff = Mean2 – Mean1   

Null Hypothesis: Mean_diff ≤ 5000  

Alternative Hypothesis: Mean_diff > 5000
        
This is a one-tail test        

#### The p-value when mean difference considered to be $5000 is 0.062

##### Power of the test 

In [ ]:
Let  Mean2 = Average sums assured by salesperson AFTER changeover. 
     Mean1 = Average sums assured by salesperson BEFORE changeover. 
     Mean_diff = Mean2 – Mean1   
        
Null Hypothesis (H0): Mean_diff = 4000

Alternative Hypothesis (HA): Mean_diff > 0

H0 will be rejected if test statistics > t_critical.

In [21]:
Standard_Error=np.std((df.iloc[:,1] - df.iloc[:,2]))/np.sqrt(len(df))
Standard_Error

2527.6251480171836

In [25]:
from scipy.stats import t
t_critical= t.ppf(1-0.05, 29)
t_critical

1.6991270265334972

In [26]:
x_bar = t_critical*Standard_Error
x_bar

4294.756201941728

In [31]:
t_stat = (x_bar - 5000)/Standard_Error
t_stat

-0.2790143936538636

In [35]:
pval = t.sf(np.abs(t_stat), 29)*2

In [36]:
pval

0.7822139200867682

With alpha = 0.05 and df = 29, critical value for t statistic (or t_critical ) will be   1.699127.

Hence, H0 will be rejected for test statistics >=  1.699127.

Hence, H0 will be rejected if for  x_bar >= 4368.176

### Power of the test 

In [33]:
print('Power of test ',smp.ttest_power(t_stat, nobs=30,alpha=0.05, alternative='two-sided'))

Power of test  0.31512058392807174
